### PR TITLE
fix CurlFactory for PHP 7.3

### PR DESCRIPTION
**Pull request by jenschude**

### CODE REVIEW

File: tests/CoreTest.php [hunk 1]
```diff
@@ -274,7 +274,7 @@ public function typeProvider()
      */
     public function testDescribesType($input, $output)
     {
-        $this->assertEquals($output, Core::describeType($input));
+        $this->assertContains($output, Core::describeType($input));
```
Comment (sagikazarmark) [inline tests/CoreTest.php]: Why is this change necessary?

**Review by sagikazarmark — COMMENTED** [review]
Looks good. I have one question I'm not entirely sure about though.

File: tests/CoreTest.php (hunk 1 above)
Comment (jenschude) [reply tests/CoreTest.php]: Because of the way PHP outputs the content with var_dump from 7.x on with information about the file and line number where the var_dump has been called. E.G.:

```
5) GuzzleHttp\Tests\Ring\CoreTest::testDescribesType with data set #4 (1.0, 'float(1)')
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
+RingPHP/src/Core.php:301:
 float(1)

RingPHP/tests/CoreTest.php:277
```

And Core::describeType uses var_dump to output information about the input

Comment (sagikazarmark) [reply tests/CoreTest.php]: I can't recall any changes to var_dump, also can't find anything in the documentation.

Also can't reproduce it here: https://3v4l.org/jXMF6

Is there a chance that it's caused by xdebug var_dump overloading? That's my guess at least.

Comment (jenschude) [reply tests/CoreTest.php]: Oh man. You are so right. The line output started for me with PHP7. But yeah it was the xdebug var_dump overload. Maybe they introduced this feature with the php7 release. I will revert this change.

**Comment by jenschude** [comment]
Btw I also could check if the setting is set with the filenames like this:

```
        if (ini_get('xdebug.overload_var_dump') == 2) {
            $this->assertContains($output, Core::describeType($input));
        } else {
            $this->assertEquals($output, Core::describeType($input));
        }
```

And build was running on my travis: https://travis-ci.org/jenschude/RingPHP/builds/409742961

**Comment by sagikazarmark** [comment]
> Btw I also could check if the setting is set with the filenames like this:

I would rather make sure that during tests that feature is off. Maybe using phpunit.xml, but for now I think it's ok as it is now.

> And build was running on my travis:

Cool

**Comment by sagikazarmark** [comment]
Thanks for the patch!

**Comment by jenschude** [comment]
Now only the patch for guzzle 5.3 is missing ;)
